tests/unit/test_compound_ids.py

```diff
@@ -25,6 +25,20 @@ class TestCompoundIds(unittest.TestCase):
     """
     Test the compound ids
     """
+    def testURLUnsafe(self):
+        hasSlashes = "???"  # base64 encodes to 'Pz8/'
+        needsPadding = "padme"  # base64 encodes to 'cGFkbWU='
+        realistic = ("YnJjYTE6V0FTSDdQX2Fubm90YXRpb246MToxNzY5"
+                     "NDpmMzQxM2JkMTVjNWNiYzI4ZDFiYjY2OGY4ZWM2NzczMg")
+        unsafeCharacters = ['$', '&', '+', ',', '/', ':', ';', '=', '?', '@']
+        for char in unsafeCharacters:
+            self.assertNotIn(char, datamodel.CompoundId.obfuscate(
+                    hasSlashes))
+            self.assertNotIn(char, datamodel.CompoundId.obfuscate(
+                    needsPadding))
+            self.assertNotIn(char, datamodel.CompoundId.obfuscate(
+                    realistic))
```

Comment:
Looks good, but we want to verify that we correctly deobfuscate ``hasSlashes`` also. This is only testing the obfuscate side of it.